File: Week01_Sep14/04_Intermediate2/StefanMendoza/python/implementation/intermediate2.py
'''
File: intermediate2.py
Author: Stefan Mendoza
Date: 4 September 2016
Description: Week 1 / Intermediate 2 for KU Programming Club (2016-2017)
'''

# To allow us to get the list of letters
import string

class LetterValueMap:
    letter_value = {}

    def __init__(self):
        self.letter_value = self.generateLetterValueMap()

    def generateLetterValueMap(self):
        mapping = {}

        for i in range(0, len(string.ascii_lowercase)):
            # Creating a mapping for both upper and lowercase letters
            mapping[string.ascii_lowercase[i]] = i + 1
            mapping[string.ascii_uppercase[i]] = i + 1

        return mapping

    def getWordValue(self, word):
        letter_value = self.generateLetterValueMap()
        value = 0

        for letter in word:
            value += letter_value[letter]

        return value

if __name__ == "__main__":
    print("What is the word you'd like to place a value on?")
    word = input("> ")

    if word == '':
        print("\nWhoops! You didn't enter anything.")
        print("Exiting...")
    else:
        value = 0
        mapping = LetterValueMap()

        try:
            value = mapping.getWordValue(word)
            print("The total value of your word is {0}.".format(value))
        except KeyError as error:
            print("\nWhoops! You entered a word with an invalid character!")
            print("Exiting...")
        except Exception as error:
            print("\nWhoops! Something went wrong.")
            print("Exiting...")

File: Week01_Sep14/04_Intermediate2/StefanMendoza/Intermediate2.cpp
/**
 * File: Intermediate2.cpp
 * Author: Stefan Mendoza
 * Date: 3 September 2016
 *
 * Week 1 / Intermediate 2 for KU Programming Club (2016-2017)
 */


#include <iostream>
#include <string>
#include <unordered_map>

using namespace std;

unordered_map<char, int>* generateletterValueMap() {
    unordered_map<char, int>* letterValueMap = new unordered_map<char, int>();

    for(int i = 65; i < 91; i++) {
        letterValueMap->insert({char(i), i - 64});
        letterValueMap->insert({char(i + 32), i - 64});
    }

    return letterValueMap;
}

int main(int argc, char** argv) {
    unordered_map<char, int>* letterValueMap = generateletterValueMap();

    std::cout << "What is the word you'd like to place a value on?" << std::endl;
    std::cout << "> ";
    string word;
    std::cin >> word;

    int value = 0;
    for(int i = 0; i < word.length(); i++) {
        if(letterValueMap->count(word[i]) == 0) {
            std::cout << "\nSorry! You entered a word with non-alphabetical characters." << std::endl;
            std::cout << "Exiting..." << std::endl;
            return 0;
        }
        else {
            value += letterValueMap->at(word[i]);
        }
    }

    std::cout << "The total value of your word is " << value << "." << std::endl;
    return 0;
}
